Xamarin.Forms.Core/BindableObjectExtensions.cs

```diff
@@ -1,10 +1,37 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Linq.Expressions;
 
 namespace Xamarin.Forms
 {
 	public static class BindableObjectExtensions
 	{
+		internal static void PropagateBindingContext<T>(this BindableObject self, IList<T> children)
+		{
+			PropagateBindingContext(self, children, BindableObject.SetInheritedBindingContext);
+		}
+
+		internal static void PropagateBindingContext<T>(this BindableObject self, IList<T> children, Action<BindableObject, object> setChildBindingContext)
+		{
+			var gotBindingContext = false;
+			object bc = null;
+
+			for (var i = 0; i < children.Count; i++)
+			{
+				var bo = children[i] as BindableObject;
```

Comment:
So it looks like we can't do this kind of pattern matching yet
https://stackoverflow.com/questions/41436399/expression-of-type-t-cannot-be-handled-by-a-pattern-of-type-x

Because children is a generic list